src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.java

```diff
@@ -688,17 +693,26 @@ private VendorDetailExtension buildVendorDetailExtension(PaymentWorksVendor pmwV
     }
     
     private List<VendorSupplierDiversity> buildVendorDiversitiesFromPmwFormDropDownLists(PaymentWorksVendor pmwVendor, Map<String, SupplierDiversity> paymentWorksToKfsDiversityMap, List<VendorSupplierDiversity> kfsVendorSupplierDiversities) {
-        VendorSupplierDiversity minority = createVendorSupplierDiversityForDropDownData(pmwVendor.getMinorityStatus(), pmwVendor.getMbeCertificationExpirationDate(), paymentWorksToKfsDiversityMap);
-        if (ObjectUtils.isNotNull(minority)) {
-            kfsVendorSupplierDiversities.add(minority);
-        }
-        VendorSupplierDiversity womanOwned = createVendorSupplierDiversityForDropDownData(pmwVendor.getWomanOwned(), pmwVendor.getWbeCertificationExpirationDate(), paymentWorksToKfsDiversityMap);
-        if (ObjectUtils.isNotNull(womanOwned)) {
-            kfsVendorSupplierDiversities.add(womanOwned);
-        }
-        VendorSupplierDiversity veteran = createVendorSupplierDiversityForDropDownData(pmwVendor.getDisabledVeteran(), pmwVendor.getVeteranCertificationExpirationDate(), paymentWorksToKfsDiversityMap);
-        if (ObjectUtils.isNotNull(veteran)) {
-            kfsVendorSupplierDiversities.add(veteran);
+        if (paymentWorksFormModeService.shouldUseLegacyFormProcessingMode()) {
+            VendorSupplierDiversity minority = createVendorSupplierDiversityForDropDownData(pmwVendor.getMinorityStatus(), 
+                    pmwVendor.getMbeCertificationExpirationDate(), paymentWorksToKfsDiversityMap);
+            if (ObjectUtils.isNotNull(minority)) {
+                kfsVendorSupplierDiversities.add(minority);
+            }
+            
+            VendorSupplierDiversity womanOwned = createVendorSupplierDiversityForDropDownData(pmwVendor.getWomanOwned(), 
+                    pmwVendor.getWbeCertificationExpirationDate(), paymentWorksToKfsDiversityMap);
+            if (ObjectUtils.isNotNull(womanOwned)) {
+                kfsVendorSupplierDiversities.add(womanOwned);
+            }
+            
+            VendorSupplierDiversity veteran = createVendorSupplierDiversityForDropDownData(pmwVendor.getDisabledVeteran(), 
+                    pmwVendor.getVeteranCertificationExpirationDate(), paymentWorksToKfsDiversityMap);
+            if (ObjectUtils.isNotNull(veteran)) {
+                kfsVendorSupplierDiversities.add(veteran);
+            }
+        } else if (paymentWorksFormModeService.shouldUseForeignFormProcessingMode()) {
+            LOG.info("buildVendorDiversitiesFromPmwFormDropDownLists, " + PaymentWorksConstants.FOREIGN_FORM_PROCESSING_NOT_IMPLEMENTED_LOG_MESSAGE);
```

Comment:
Would it make sense to have an empty stub method here for processing in case of foreign form mode?